haystack/errors.py

```diff
@@ -181,6 +181,13 @@ def __init__(self, message: Optional[str] = None):
         super().__init__(message=message)
 
 
+class HuggingFacePipelineInferenceError(NodeError):
```

Comment:
`HuggingFacePipelineError` sounds better to me...
Inference reminds me of the HF Inference API...